pgvectorscale/src/access_method/graph/neighbor_store.rs

```diff
@@ -1,76 +1,146 @@
-use std::collections::BTreeMap;
+use std::cell::RefCell;
+use std::num::NonZero;
 
+use lru::LruCache;
+use pgrx::warning;
+
+use crate::access_method::build::maintenance_work_mem_bytes;
 use crate::util::{IndexPointer, ItemPointer};
 
 use crate::access_method::graph::neighbor_with_distance::*;
 use crate::access_method::labels::LabelSet;
 use crate::access_method::meta_page::MetaPage;
-use crate::access_method::stats::{StatsDistanceComparison, StatsNodeModify, StatsNodeRead};
+use crate::access_method::stats::PruneNeighborStats;
 use crate::access_method::storage::Storage;
 
+use super::Graph;
+
 /// A builderGraph is a graph that keep the neighbors in-memory in the neighbor_map below
 /// The idea is that during the index build, you don't want to update the actual Postgres
 /// pages every time you change the neighbors. Instead you change the neighbors in memory
 /// until the build is done. Afterwards, calling the `write` method, will write out all
 /// the neighbors to the right pages.
-#[derive(Default)]
+///
+pub struct NeighborCacheEntry {
+    pub labels: Option<LabelSet>,
+    pub neighbors: Vec<NeighborWithDistance>,
+}
+
+impl NeighborCacheEntry {
+    pub fn new(labels: Option<LabelSet>, neighbors: Vec<NeighborWithDistance>) -> Self {
+        Self { labels, neighbors }
+    }
+
+    /// Estimate of the size of an entry in the cache in bytes.
+    pub fn size(num_neighbors: usize, has_labels: bool) -> usize {
+        std::mem::size_of::<Self>()
+            + num_neighbors * std::mem::size_of::<NeighborWithDistance>()
+            + if has_labels {
+                std::mem::size_of::<LabelSet>() + 4
+            } else {
+                0
+            }
```

Comment:
This shouldn't be added; the size of the `LabelSet` is already included in `std::mem::size_of::<Self>()` (the size of an `Option<T>` is at least the size of `T`).